Add a check on minQ <= maxQ

So-Fras opened the pull request:
**Please check if the PR fulfills these requirements**
<!-- please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes -->
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?**
Quality

**Does this PR introduce a new Powsybl Action implying to be implemented in simulators or pypowsybl?**
- [ ] Yes
- [x] No

**What is the current behavior?**
There is no check to ensure minQ <= maxQ in a ReactiveCapabilityCurve Point.


**What is the new behavior (if this is a feature change)?**
In `PointImpl`, a constructor is added to allow de/activating a new check is added to ensure `minQ <= maxQ`. An exception is thrown when the check is activated and `minQ > maxQ`.
The existing constructor will now call the new one with the check activated.

The same check has been added to the `PointAdderImpl`, in `ReactiveCapabilityCurveAdderImpl`.
A condition has been added to deactivate the check for IIDM versions preceding the version 1.14.

**Does this PR introduce a breaking change or deprecate an API?**
- [x] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**
<!-- If no breaking changes or API deprecations were introduced, delete this section -->
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**
<!-- If this PR introduces a breaking change, describe the migration steps -->
<!-- The content of this section will be copied in the migration guide -->
If you need to create a `PointImpl` with `minQ > maxQ`, you need to change `PointImpl(p, minQ, maxQ)` to `PointImpl(p, minQ, maxQ, false)`

Custom IIDM implementations' maintainers should implement `ReactiveCapabilityCurveAdder.setShouldCheckMinMaxValues(boolean shouldCheckMinMaxValues)`.



**Other information**:
- Is it really a change in IIDM since no new attribute is added? Maybe it is since an IIDM file with a minQ > maxQ will not be processed the same way as before...
- Should we throw an exception or should we "fix" one of the values (take the smallest value both for minQ and maxQ to be conservative?) with a warning to make the model more robust?

So-Fras committed:
Add a check on Qmin <= Qmax

Signed-off-by: Sophie Frasnedo <sophie.frasnedo@rte-france.com>
Signed-off-by: Nicolas Rol <nicolas.rol@rte-france.com>

geofjamg commented:
Yes this is a breaking change as you won't be able to load old file without any exception. You can add this kind of change BUT only in a new version of IIDM and keeping old behaviour (do nothing) for other IIDM versions.

flo-dup commented on iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ReactiveCapabilityCurveImpl.java:
```diff
@@ -30,6 +30,9 @@ static class PointImpl implements Point {
         private double maxQ;
 
         PointImpl(double p, double minQ, double maxQ) {
+            if (minQ > maxQ) {
+                throw new IllegalStateException("minQ should be inferior or equal to maxQ");
+            }
```
There's a related TODO written in 2017 (to plan it for IIDM v1.1!!) in the corresponding adder, and a `@Disabled` in a unit test in AbstractReactiveCapabilityCurveTest. Could you remove them?

This should be only activated for next IIDM version. There should be a check in the importer based on the IIDM version to flip or not.

rolnico committed:
Add validation for minQ/maxQ in reactive capability curves

Introduce a mechanism to validate the relationship between minQ and maxQ, ensuring maxQ is not less than minQ. This check is enabled by default starting from IIDM v1.14 but remains optional for earlier versions to maintain compatibility. Corresponding tests and serialization adjustments have been added to ensure reliability.

Signed-off-by: Nicolas Rol <nicolas.rol@rte-france.com>

rolnico committed:
Add test on setShouldCheckMinMaxValues

Signed-off-by: Nicolas Rol <nicolas.rol@rte-france.com>

rolnico committed:
add javadoc on setShouldCheckMinMaxValues

Signed-off-by: Nicolas Rol <nicolas.rol@rte-france.com>

geofjamg reviewed (changes requested):
I don't think this is a good idea to have a option to enable or not the check. We should just add the check and fix minq and maxq in importers (like just inverting)

So-Fras commented:
Newer PR opened on this topic #3617 that mentions this PR.
This one can be closed.
